tests/cbtest/gtest/test_zrepl_prot.cc

```diff
@@ -1020,7 +1020,52 @@ TEST_F(ZreplDataTest, RebuildFlag) {
 	ASSERT_EQ(rc, sizeof (buf));
 
 	/* read the block with rebuild flag */
-	read_data_start(m_data_fd1, m_ioseq1, 0, sizeof (buf), &hdr_in, true);
+	read_data_start(m_data_fd1, m_ioseq1, 0, sizeof (buf), &hdr_in, ZVOL_OP_FLAG_REBUILD);
+	ASSERT_EQ(hdr_in.status, ZVOL_OP_STATUS_OK);
+	ASSERT_EQ(hdr_in.len, sizeof (read_hdr) + sizeof (buf));
+	rc = read(m_data_fd1, &read_hdr, sizeof (read_hdr));
+	ASSERT_ERRNO("read", rc >= 0);
+	ASSERT_EQ(rc, sizeof (read_hdr));
+	ASSERT_EQ(read_hdr.io_num, 654);
+	ASSERT_EQ(read_hdr.len, sizeof (buf));
+	rc = read(m_data_fd1, buf, sizeof (buf));
+	ASSERT_ERRNO("read", rc >= 0);
+	ASSERT_EQ(rc, sizeof (buf));
+}
+
+/*
+ * Metadata ionum should be returned when equested by
+ * ZVOL_OP_FLAG_READ_METADATA flag.
+ */
+TEST_F(ZreplDataTest, ReadMetaDataFlag) {
+	zvol_io_hdr_t hdr_in, hdr_out;
+	struct zvol_io_rw_hdr read_hdr;
+	struct zvol_io_rw_hdr write_hdr;
+	struct zrepl_status_ack status;
+	struct mgmt_ack mgmt_ack;
+	char buf[4096];
+	int rc;
+
+	/* write a data block with known ionum */
+	write_data_and_verify_resp(m_data_fd1, m_ioseq1, 0, 654);
+
+	sleep(5);
```

Comment:
we should avoid sleeps with magic constants in our tests which are inherently unreliable. If it is necessary then we need some comment and explanation why is that here.

Reply:
ok. I think we can avoid sleep here. I will remove sleep.